libraries/freertos_plus/standard/freertos_plus_cli/source/FreeRTOS_CLI_Console.c

```diff
@@ -0,0 +1,178 @@
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "FreeRTOS_CLI_Console.h"
+
+#define cmdMAX_INPUT_BUFFER_SIZE    50
+
+#define cmdMAX_ERROR_SIZE           50
+
+/* DEL acts as a backspace. */
+#define cmdASCII_DEL                ( 0x7F )
+
+#define cmdERROR_DELAY              ( 1000 )
+
+/* Const messages output by the command console. */
+static const char * const pcStartMessage = "\r\n[Input a command or type help then press ENTER to get started]\r\n>";
+static const char * const pcEndOfOutputMessage = "\r\n[Press Enter to run the previous command]\r\n>";
+static const char * const pcNewLine = "\r\n";
+
+static char cInputBuffer[ cmdMAX_INPUT_BUFFER_SIZE ] = "";
+static char cErrorString[ cmdMAX_ERROR_SIZE ] = "";
+
+static uint8_t ucCommandIndex = 0;
+
+static void processInputBuffer( xConsoleIO_t consoleIO,
+                                int32_t inputSize,
+                                char * pCommandBuffer,
+                                size_t commandBufferLength,
+                                char * pOutputBuffer,
+                                size_t outpuBufferLength );
+
+void FreeRTOS_CLIEnterConsoleLoop( xConsoleIO_t consoleIO,
+                                   char * pCommandBuffer,
+                                   size_t commandBufferLength,
+                                   char * pOutputBuffer,
+                                   size_t outputBufferLength )
+{
+    BaseType_t status = pdTRUE;
+    int32_t bytesRead;
+
+    if( ( consoleIO.read == NULL ) || ( consoleIO.write == NULL ) )
+    {
+        status = pdFALSE;
+    }
+
+    if( status == pdTRUE )
+    {
+        if( ( pCommandBuffer == NULL ) || ( commandBufferLength == 0 ) )
+        {
+            status = pdFALSE;
+        }
+    }
+
+    if( status == pdTRUE )
```

Comment:
Same here

Reply:
Replaced with `configASSERT` since these parameters are not going to change during the runtime of the system.